### PR TITLE
feat: updated the README.md for anymaths; fixed AnyMathsAdapter code; provided example training and dataset preparation

### PR DESCRIPTION
**Pull request by egmaminta**
#### Summary
This PR improves the `AnyMaths` example end-to-end by fixing the adapter code (I found a bug that only outputs 1 Pareto val set), adding dataset preparation utilities, providing a runnable sample training script, and updating the `AnyMaths` README with clear usage guidance.

#### Changes
- Docs
    - Updated `README.md` with setup, configuration, and run instructions.
- Adapter
    - Fixed `AnyMaths` adapter to correctly handle multi-item evaluation (no more assertion error when valset > 1).
    - Ensured batched/iterative evaluation returns consistent metrics and aggregates correctly.
- Dataset preparation
    - Added dataset loading and splitting logic supporting:
        - `"openai/gsm8k"` (train/test) with deterministic shuffling and derived solutions.
        - `"MathArena/aime_2025"` with aligned train/test preprocessing.
    - Provides train/val split from train; expands a small test set for demo runs.
- Sample training
    - Added `train_anymaths.py`:
        - CLI args for model, API base, budget, reflection LM, minibatch size, and seed.
        - Uses `gepa.optimize` with `AnyMathsAdapter`.
        - LiteLLM-based reflection LM hook.
        - Small default subset sizes for quick demos.

#### Why
- The adapter previously triggered an assertion when `valset > 1`. This is non-ideal for real validation and made experiments brittle.
- Providing dataset prep and a runnable script lowers the barrier to trying AnyMaths with GEPA.

#### How to use
- Start an Ollama server (or compatible endpoint) and ensure the requested models are available.
- Run the training script (example):
    - Module: `python -m gepa.examples.anymaths-bench.train_anymaths --model_name ollama/qwen3:4b --api_base http://localhost:11434 --anymaths_dset_name openai/gsm8k --budget 50`
- Adjust `--reflection_lm`, `--reflection_minibatch_size`, and `--seed` as needed.

#### Notes
- Defaults limit train/val/test to 10 items for fast demos; expand as desired.
- No breaking API changes expected outside the AnyMaths example path.

Checklist

- [x] Docs updated
- [x] Adapter supports multi-item validation
- [x] Dataset prep utilities included
- [x] Sample training script added and runnable

### CODE REVIEW

File: src/gepa/examples/anymaths-bench/prompt-templates/instruction_prompt.txt
Comment (LakshyAAAgrawal) [inline src/gepa/examples/anymaths-bench/prompt-templates/instruction_prompt.txt:1]: Is this meant to be used as the starter prompt?

Comment (egmaminta) [reply src/gepa/examples/anymaths-bench/prompt-templates/instruction_prompt.txt:1]: Yes but as a template! A user may modify it.

**Comment by LakshyAAAgrawal** [comment]
Hey, can you include any experiment results in the README?
